python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -72,7 +73,13 @@ class TorchBasedFeature(Feature):
     device(type='cuda', index=0)
     """
 
-    def __init__(self, torch_feature: torch.Tensor, metadata: Dict = None):
+    def __init__(
```

Comment:
I don't quite like that TorchBasedFeature is also the DiskBasedFeature. I feel the two should be seperated. @frozenbugs what do you think?

Reply:
It feels like a hack. If our abstractions are right, we should not need any hacks like this.

Reply:
I agree, we should introduce a new DiskBasedFeature instead of keep overloading TorchBasedFeature.